### PR TITLE
Add a new error code 8129

### PR DESCRIPTION
**Pull request by dyzsr**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

Add a new error-code to TiDB for https://github.com/pingcap/tidb/issues/21779.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by XuHuaiyu** [comment]
LGTM

File: error-codes.md [hunk 1]
```diff
@@ -331,6 +331,10 @@ In addition, TiDB has the following unique error codes:
     
     Modify the SQL statement or disable the `only_full_group_by` mode.
 
+* Error Number: 8129
+
+    TiDB doesn't yet support JSON object with key length >= 65536.
```
Comment (TomShawn) [inline error-codes.md:336]:
```suggestion
    TiDB does not yet support JSON objects with the key length >= 65536.
```

**Review by TomShawn — APPROVED** [review]
LGTM

**Comment by TomShawn** [comment]
@dyzsr Will this PR be cherry-picked to release-4.0?

**Comment by dyzsr** [comment]
> @dyzsr Will this PR be cherry-picked to release-4.0?

I'm not sure about this. @XuHuaiyu

**Comment by dyzsr** [comment]
> @dyzsr Will this PR be cherry-picked to release-4.0?

Yes. We have to cherry-pick to 4.0

**Pull request merged**